File: Backend/AstroGame.Shared/Models/Technologies/PlayerDependentTechnologyUpgrade.cs
using System;
using AstroGame.Shared.Models.Players;
using Newtonsoft.Json;

namespace AstroGame.Shared.Models.Technologies
{
    public class PlayerDependentTechnologyUpgrade : TechnologyUpgrade
    {
        public Guid PlayerId { get; set; }
      
        [JsonIgnore] public virtual Player Player { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/technologies/player_dependent_technology_upgrade.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/technologies/technology_upgrade.dart';

import 'package:json_annotation/json_annotation.dart';

part 'player_dependent_technology_upgrade.g.dart';

@GuidConverter()
@JsonSerializable()
class PlayerDependentTechnologyUpgrade extends TechnologyUpgrade {
  late Guid playerId;

  PlayerDependentTechnologyUpgrade();

  factory PlayerDependentTechnologyUpgrade.fromJson(
          Map<String, dynamic> json) =>
      _$PlayerDependentTechnologyUpgradeFromJson(json);
  Map<String, dynamic> toJson() =>
      _$PlayerDependentTechnologyUpgradeToJson(this);
}
